hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/CapacityScheduler.java

```diff
@@ -455,8 +458,15 @@ public void reinitialize(Configuration newConf, RMContext rmContext,
         refreshMaximumAllocation(
             ResourceUtils.fetchMaximumAllocationFromConfig(this.conf));
         reinitializeQueues(this.conf);
+        reinitRequestsHandler(this.conf);
       } catch (Throwable t) {
         this.conf = oldConf;
+        try {
+          reinitRequestsHandler(this.conf);
+        } catch (Throwable innerT) {
+          LOG.error("Failed to re-init requests handler : {}",
```

Comment:
Will the basic functions be affected if the initialization fails?

Reply:
Thanks @zeekling  for the review. 
oldConf is a configuration that has already been validated and is in effect, the invocation of CapacityScheduler#reinitRequestsHandler here will not throw an exception actually. The purpose of this try/catch block is only included to make sure that the original rollback logic remains unaffected.

Reply:
got it